CRM/Core/BAO/CustomField.php

```diff
@@ -148,6 +148,9 @@ public static function bulkSave($bulkParams, $defaults = []) {
       Civi::service('sql_triggers')->rebuild($params['table_name'], TRUE);
     }
     CRM_Utils_System::flushCache();
+    // Flush caches is not aggressive about clearing the specific cache we know we want to clear
+    // so do it manually. Ideally we wouldn't need to clear others...
+    Civi::cache('metadata')->clear();
     foreach ($customFields as $index => $customField) {
```

Comment:
@colemanw it DOES seem to be in there

```
    if (Civi\Core\Container::isContainerBooted()
      && !in_array(get_class(CRM_Utils_Cache::singleton()), $localDrivers)) {
      Civi::cache('long')->flush();
      Civi::cache('settings')->flush();
      Civi::cache('js_strings')->flush();
      Civi::cache('community_messages')->flush();
      Civi::cache('groups')->flush();
      Civi::cache('navigation')->flush();
      Civi::cache('customData')->flush();
      Civi::cache('contactTypes')->clear();
      Civi::cache('metadata')->clear();
      CRM_Extension_System::singleton()->getCache()->flush();
      CRM_Cxn_CiviCxnHttp::singleton()->getCache()->flush();
    }
```

I was wondering if we could ONLY flush metadata & maybe fields

Reply:
@eileenmcnaughton ok I just copied & pasted it here from the `\CRM_Core_BAO_CustomField::create()` function.

Reply:
@colemanw so it might be worth articulating how we think it should work. The way I see it is 'metadata' should be the cache for things that only change when there is a site config change - ie a field change, a membership type, a price set - for most sites these things aren't daily.  I would probably merge 'contactTypes' & 'fields' into metadata 

The other caches that could be affected are pseudoconstants - mostly we can just unset Civi::statics for now I guess but ideally they would be metadata over time (metadata accesses from cache backend once & an array cache for the rest of the request)

Reply:
@eileenmcnaughton I've taken that bit out since it's not a blocker to this PR.

Reply:
thanks - it looks generally OK to me - hopefully @highfalutin can test